### PR TITLE
Use exceptions for validation functions

### PR DESCRIPTION
**Pull request by clark800**

### CODE REVIEW

File: api/lib/validate.js [hunk 1]
```diff
@@ -1,99 +1,137 @@
...
+    throw error('Secret does not match specified address', secret);
   }
-  return null;
 }
 
 function validateCurrency(currency) {
   if (!validator.isValid(currency, 'Currency')) {
-    return new errors.InvalidRequestError(
-      'Parameter is not a valid currency: currency');
+    throw error('Parameter is not a valid currency: currency');
```
Comment (lumberj) [inline api/lib/validate.js]: Did you mean to uncomment the next line and delete this one?

File: api/lib/validate.js [hunk 2]
```diff
@@ -1,99 +1,137 @@
...
   }
-  return null;
 }
 
 function validateCounterparty(counterparty) {
   if (!ripple.UInt160.is_valid(counterparty)) {
-    return new errors.InvalidRequestError(
-      'Parameter is not a valid Ripple address: counterparty');
+    throw error('Parameter is not a valid Ripple address: counterparty');
+    // throw invalid('counterparty', counterparty);
```
Comment (lumberj) [inline api/lib/validate.js]: Same as above

File: api/lib/validate.js (hunk 1 above)
Comment (clark800) [inline api/lib/validate.js]: No, the commented ones are TODOs. I don't want to change all the test cases yet.

Comment (lumberj) [inline api/lib/validate.js]: Can you write `TODO` in the comment to make it more clear? Or are you working on this right now?

Comment (clark800) [inline api/lib/validate.js]: Ok, updated with `TODO`s

**Comment by lumberj** [comment]
LGTM

**Pull request merged**